docker-for-mac/multi-arch.md

```diff
@@ -54,3 +54,162 @@ Notice that this time, the `uname -a` output shows `armv7l` and
+
+Docker Desktop 2.0.4.0 Edge release introduces a new CLI command called `buildx`.  Buildx allows you to locally build multi-arch images, link them together with a manifest file, and push them all to a registry using a single command.  With the included emulation, you can transparently build more than just native images.  Buildx accomplishes this by adding new builder instances based on BuildKit, and leveraging Docker Desktop's technology stack to run non-native binaries.
+
+### Install
+
+1. Download Docker Desktop 2.0.4.0 or higher.
+
+    - [Docker Desktop 2.0.4.0 for Mac](https://docs.docker.com/docker-for-mac/edge-release-notes/)
+    - [Docker Desktop 2.0.4.0 for Windows](https://docs.docker.com/docker-for-windows/edge-release-notes/)
+
+1. Follow the on-screen instructions to complete the installation process. After you have successfully installed Docker Desktop, you will see the Docker icon in your task tray.
+
+1. Click **About Docker Desktop** from the Docker menu and ensure you have installed Docker Desktop version 2.0.4.0 (33772) or higher.
+
+![about-docker-desktop-buildx](./images/desktop-buildx-version.png)
+
+### Build and run multi-architecture images
+
+Run the command `docker buildx ls` to list the existing builders. This displays the default builder, which is our old builder.
+
+```
+~ ❯❯❯ docker buildx ls
+NAME/NODE DRIVER/ENDPOINT STATUS  PLATFORMS
+default * docker
+  default default         running linux/amd64, linux/arm64, linux/arm/v7, linux/arm/v6
+
+```
+
+Create a new builder which gives access to the new multi-architecture features.
+
+```
+~ ❯❯❯ docker buildx create --name mybuilder
+mybuilder
+```
+
+Switch to the new builder and inspect it.
+
+```
+~ ❯❯❯ docker buildx use mybuilder`
+~ ❯❯❯ docker buildx inspect --bootstrap
+[+] Building 2.5s (1/1) FINISHED
+ => [internal] booting buildkit                                                   2.5s
+ => => pulling image moby/buildkit:master                                         1.3s
+ => => creating container buildx_buildkit_mybuilder0                              1.2s
+Name:   mybuilder
+Driver: docker-container
+
+Nodes:
+Name:      mybuilder0
+Endpoint:  unix:///var/run/docker.sock
+Status:    running
+
+Platforms: linux/amd64, linux/arm64, linux/arm/v7, linux/arm/v6
+```
+
+Test the workflow to ensure you can build, push, and run multi-architecture images. Create a simple example Dockerfile, build a couple of image variants, and push them to Docker Hub.
+
+```
+~ ❯❯❯ mkdir test && cd test && cat <<EOF > Dockerfile
+FROM ubuntu
+RUN apt-get update && apt-get install -y curl
+WORKDIR /src
+COPY . .
+EOF
+```
+
+```
+~/test ❯❯❯ docker buildx build --platform linux/amd64,linux/arm64,linux/arm/v7 -t username/demo:latest --push .
```

Comment:
You should call out that username here needs to be a valid user.

Reply:
Added a note to convey that 'username' must be a valid Docker username.